Makefile

```diff
@@ -7,14 +7,63 @@ pytest/install:
 	@ $(MAKE) install/pip/$(@D) PYPI_PKG_NAME=$(@D)
 	@ python -m pip install -r lambda/tests/requirements_dev.txt
 
+.PHONY: python/deps
 python/deps:
 	@ echo "[$@] Installing package dependencies"
 	@ python -m pip install -r lambda/src/requirements.txt
 
+.PHONY: python/test
 python/test: | guard/program/pytest
 python/test:
 	@ echo "[$@] Starting Python tests"
-	pytest
+	pytest lambda/tests
 	@ echo "[$@]: Tests executed!"
 
-.PHONY: python/deps python/test
+.PHONY: terraform/terratest
+terraform/terratest: TIMEOUT ?= 20m
+terraform/terratest:| guard/program/terraform guard/program/go
+	@ echo "[$@] Starting Terratest against lambda installation"
+	@ echo "[$@] Terraform 'apply' command is slow ... be patient !!!"
+	cd $(TERRAFORM_TEST_DIR) && go test -count=1 -timeout $(TIMEOUT)
+	@ echo "[$@]: Completed successfully!"
+
+.PHONY: terratest/setup
+terratest/setup:
+	@ echo "[$@] Create go.mod, go.sum files with updated dependencies"
+	go version
+	cd $(TERRAFORM_TEST_DIR) && rm -f go.mod go.sum
+	cd $(TERRAFORM_TEST_DIR) && go mod init tardigrade-ci/tests
+	cd $(TERRAFORM_TEST_DIR) && go mod tidy
```

Comment:
The line that follows the `go mod tidy` (on line 36) edits `go.mod` to add the 'replace' for the Terratest patch.  Once you add the "replace", you need to run `go mod tidy` again.  When we no longer need the Terratest patch, we will no longer need the second `go mod tidy`.

Reply:
Right, but this is what I have in `go.mod` when I remove the first `go mod tidy`, and the test runs fine...

```
module tardigrade-ci/tests

go 1.15

replace github.com/gruntwork-io/terratest => github.com/ffernandezcast/terratest v0.28.6-0.20201201084725-13e8a4c156b8

require (
        github.com/gruntwork-io/terratest v0.0.0-00010101000000-000000000000
        github.com/stretchr/testify v1.7.0
)
```

Reply:
I see what you mean -- you did the replace before there was even a required module to be replaced.  I believe the "standard format" for go.mod is to have the replace at the bottom of the file.  

So yes it works, but of two icky approaches, I prefer the one I have because it's logically replacing an existing module and the "replace" is then located where it should be.